db/migrate/20170112175131_migrate_template_to_parameters_macros.rb

```diff
@@ -0,0 +1,53 @@
+class MigrateTemplateToParametersMacros < ActiveRecord::Migration
+  def up
+    Template.descendants.each do |klass|
+      klass.all.each do |template|
```

Comment:
it needs to be `unscoped`, otherwise taxonomy (or user) filter leads to return an empty set and nothing actually gets migrated.